stratum/hal/lib/bcm/bcm_node.cc

```diff
@@ -318,17 +319,28 @@ ::util::Status BcmNode::ReadForwardingEntries(
   return ::util::OkStatus();
 }
 
-::util::Status BcmNode::RegisterPacketReceiveWriter(
-    const std::shared_ptr<WriterInterface<::p4::v1::PacketIn>>& writer) {
+::util::Status BcmNode::RegisterStreamMessageResponseWriter(
+    const std::shared_ptr<WriterInterface<::p4::v1::StreamMessageResponse>>&
+        writer) {
   absl::WriterMutexLock l(&lock_);
   if (!initialized_) {
     return MAKE_ERROR(ERR_NOT_INITIALIZED) << "Not initialized!";
   }
+  auto w =
+      std::make_shared<ConstraintChannelWriter<::p4::v1::StreamMessageResponse,
+                                               ::p4::v1::PacketIn>>(
+          writer, &::p4::v1::StreamMessageResponse::mutable_packet);
```

Comment:
This creates a wrapper around the generic `StreamMessageResponse` writer that only accepts `PacketIn` messages. Since it satisfies the `WriterInterface<::p4::v1::PacketIn>` interface, we can pass it to the unmodified managers and maintain type safety.